google-oauth-client/src/main/java/com/google/api/client/auth/oauth2/JWTAuthentication.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) 2021 Jun Ying.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.api.client.auth.oauth2;
+
+import com.google.api.client.http.HttpExecuteInterceptor;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpRequestInitializer;
+import com.google.api.client.http.UrlEncodedContent;
+import com.google.api.client.util.Data;
+import com.google.api.client.util.Preconditions;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * Client credentials specified as URL-encoded parameters in the HTTP request body as specified in
+ * <a href="https://tools.ietf.org/html/rfc7523">JSON Web Token (JWT) Profile
+ *       for OAuth 2.0 Client Authentication and Authorization Grants</a>
+ *
+ * <p>This implementation assumes that the {@link HttpRequest#getContent()} is {@code null} or an
+ * instance of {@link UrlEncodedContent}. This is used as the client authentication in {@link
+ * TokenRequest#setClientAuthentication(HttpExecuteInterceptor)}.
+ *
+ * <p>
+ *     To use JWT authentication, grant_type must be "client_credentials".
+ *     If AuthorizationCodeTokenRequest.setGrantType() is called, set it to
+ *     JWTAuthentication.GRANT_TYPE_CLIENT_CREDENTIALS. It can also be left
+ *     uncalled. Setting it to any other value will cause an IllegalArgumentException.
```

Comment:
will cause --> causes (Google style is present tense)

Reply:
Done.